AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -79,120 +89,101 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
+        guard storageAdapter != nil else {
             log.error(error: DataStoreError.nilStorageAdapter())
             remoteSyncTopicPublisher.send(completion: .failure(DataStoreError.nilStorageAdapter()))
             return
         }
-        remoteSyncTopicPublisher.send(.storageAdapterAvailable)
-
-        let pauseSubscriptionsOp = CancelAwareBlockOperation {
-            self.pauseSubscriptions()
-        }
-
-        let pauseMutationsOp = CancelAwareBlockOperation {
-            self.pauseMutations()
-        }
-        pauseMutationsOp.addDependency(pauseSubscriptionsOp)
 
-        let setUpCloudSubscriptionsOp = CancelAwareBlockOperation {
-            self.setUpCloudSubscriptions(api: api, storageAdapter: storageAdapter)
-        }
-        setUpCloudSubscriptionsOp.addDependency(pauseMutationsOp)
-
-        let performInitialQueriesOp = CancelAwareBlockOperation {
-            self.performInitialQueries()
-        }
-        performInitialQueriesOp.addDependency(setUpCloudSubscriptionsOp)
-
-        let activateCloudSubscriptionsOp = CancelAwareBlockOperation {
-            self.activateCloudSubscriptions()
-        }
-        activateCloudSubscriptionsOp.addDependency(performInitialQueriesOp)
-
-        let startMutationQueueOp = CancelAwareBlockOperation {
-            self.startMutationQueue(api: api, mutationEventPublisher: self.mutationEventPublisher)
-        }
-        startMutationQueueOp.addDependency(activateCloudSubscriptionsOp)
-
-        let updateStateOp = CancelAwareBlockOperation {
-            Amplify.Hub.dispatch(to: .dataStore,
-                                 payload: HubPayload(eventName: HubPayload.EventName.DataStore.syncStarted))
-            self.remoteSyncTopicPublisher.send(.syncStarted)
-        }
-        updateStateOp.addDependency(startMutationQueueOp)
-
-        syncQueue.addOperations([
-            pauseSubscriptionsOp,
-            pauseMutationsOp,
-            setUpCloudSubscriptionsOp,
-            performInitialQueriesOp,
-            activateCloudSubscriptionsOp,
-            startMutationQueueOp,
-            updateStateOp
-        ], waitUntilFinished: false)
+        remoteSyncTopicPublisher.send(.storageAdapterAvailable)
+        stateMachine.notify(action: .receivedStart)
     }
 
     func submit(_ mutationEvent: MutationEvent) -> Future<MutationEvent, DataStoreError> {
         return mutationEventIngester.submit(mutationEvent: mutationEvent)
     }
 
     // MARK: - Startup sequence
-
     private func pauseSubscriptions() {
         log.debug(#function)
         reconciliationQueue?.pause()
+
+        remoteSyncTopicPublisher.send(.subscriptionsPaused)
+        stateMachine.notify(action: .pausedSubscriptions)
     }
 
     private func pauseMutations() {
         log.debug(#function)
         outgoingMutationQueue.pauseSyncingToCloud()
+
         remoteSyncTopicPublisher.send(.mutationsPaused)
+        if let api = self.api, let storageAdapter = self.storageAdapter {
```

Comment:
Is there an error condition that pertains to not having an API or storageAdapter?

Reply:
There is a guard/else for storage adapter.  Inside of the start function, if the storageAdapter is available, we send:
```
        remoteSyncTopicPublisher.send(.storageAdapterAvailable)
```
In a subsequent PR I can change this so that we can check for both storage adapter and api and send status up that both of these are available (or perhaps that we are ready to start the sync engine)

Reply:
Updated the code to set self.api = api after checking for a nil storage adapter.  api in the context that i'm referring to is guaranteed to be not nil.  See PR: https://github.com/aws-amplify/amplify-ios/pull/329